File: app/src/main/java/fr/free/nrw/commons/wikidata/mwapi/MwResponse.java
package fr.free.nrw.commons.wikidata.mwapi;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import fr.free.nrw.commons.wikidata.json.PostProcessingTypeAdapter;
import fr.free.nrw.commons.wikidata.model.BaseModel;

import java.util.List;

public abstract class MwResponse extends BaseModel implements PostProcessingTypeAdapter.PostProcessable {
    @SuppressWarnings({"unused"}) @Nullable private List<MwServiceError> errors;
    @SuppressWarnings("unused,NullableProblems") @SerializedName("servedby") @NonNull private String servedBy;

    @Override
    public void postProcess() {
        if (errors != null && !errors.isEmpty()) {
            throw new MwException(errors.get(0), errors);
        }
    }
}

File: app/src/main/java/fr/free/nrw/commons/wikidata/mwapi/MwResponse.kt
package fr.free.nrw.commons.wikidata.mwapi

import com.google.gson.annotations.SerializedName
import fr.free.nrw.commons.wikidata.json.PostProcessingTypeAdapter.PostProcessable
import fr.free.nrw.commons.wikidata.model.BaseModel

abstract class MwResponse : BaseModel(), PostProcessable {
    private val errors: List<MwServiceError>? = null

    @SerializedName("servedby")
    private val servedBy: String? = null

    override fun postProcess() {
        if (!errors.isNullOrEmpty()) {
            throw MwException(errors[0], errors)
        }
    }
}
